### PR TITLE
Quartermaster HalJson API

### LINKED ISSUE #31: implement quartermaster APIs
implement the following json APIs using **mocked data**:
* `api/dashboards` - GET - fetch all configured dashboard ids
* `api/dashboards/{dashboardId}` - GET - fetch the specified dashboard including all widgets configurations for this dashboard
* `api/widgets/{widgetId}` - GET - fetch widget config for specified widget
* `api/widgets/{widgetId}` - POST - create a widget 
* `api/widgets/{widgetId}/data` - GET - fetch the current widget data for specified widget

example json to create a widget:
```json
{
  "title" : "myUrlWidget",
  "sourceConfigs" : [
    {
      "id" : "urlSource",
      "type" : "urlSource",
      "interval" : 1000,
      "configData" :
        {
          "url" : "https://echo.getpostman.com/get?test=123"
        }
     }
  ]
}
```

Widget IDs are globally unique

### LINKED ISSUE #35: write HalJson Renderer
As Spring HATEOAS renders not nicely (links vs. _links, no embedded relation) a custom HalJson representation will be written for infiniboard APIs.

HalJson Draft: https://tools.ietf.org/html/draft-kelly-json-hal-08#section-4.1

### LINKED ISSUE #55: fix mongo host for Docker container
Currently the docker containers are trying to connect to `localhost`.

The containers should be able to override the application.properties and connect to host `mongo`.

### PR DESCRIPTION
**Pull request by matthiasbalke**
This PR implements a basic HalJson API for quartermaster.

Features
- [x] added endpoint `/api/dashboards` (closes #31)
- [x] added endpoint `/api/dashboards/<id>`
- [x] integrated `WidgetConfig` into `Dashboard`
- [x] added links between HalJson resources
- [x] renders JSON as HalJson

Libraries
- [x] initial version of HalJsonResource (closes #35)
- [x] updated Spring Loaded to version `1.2.5`

Development
- [x] CORS Support for http://localhost:3000

Build
- [x] docker containers are using a custom application config now (closes #55)
- [x] harvester: fixed ports exported by docker container
- [x] harvester: fixed server.port use by docker container
- [x] harvester: using health check to verify linkage between docker containers
- [x] quartermaster: added actuator for health checking

### CODE REVIEW

**Comment by matthiasbalke** [comment]
Work in progress. Do not merge yet.

**Comment by matthiasbalke** [comment]
ready for merge

File: quartermaster/src/main/java/com/github/reflectoring/Quartermaster.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        registrationBean.addInitParameter("confPath", "urlrewrite.xml");
+//        registrationBean.addInitParameter("statusEnabled", "true");
+//        registrationBean.addInitParameter("logLevel", "DEBUG");
+
+        return registrationBean;
+    }
+
+    @Bean
+    public WebMvcConfigurer corsConfigurer() {
+        return new WebMvcConfigurerAdapter() {
```
Comment (Pentadrago) [inline quartermaster/src/main/java/com/github/reflectoring/Quartermaster.java]: some comments would be nice

File: quartermaster/src/main/java/com/github/reflectoring/infiniboard/quartermaster/dashboard/WidgetData.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    public static HalJsonResource toResource(Dashboard dashboard, WidgetData widgetData) {
+        HalJsonResource resource = new HalJsonResource();
+
+        resource.add("dashboardId", widgetData.getDashboardId());
+        resource.add("widgetId", widgetData.getWidgetId());
+        resource.add("data", widgetData.getData());
+
+        resource.add(new Link("self", "http://localhost:8090/api/dashboards/" + dashboard.getId() + "/widget-data/" + widgetData.getWidgetId()));
```
Comment (Pentadrago) [inline quartermaster/src/main/java/com/github/reflectoring/infiniboard/quartermaster/dashboard/WidgetData.java]: wrong port, correct one is 8080

File: quartermaster/src/main/java/com/github/reflectoring/Quartermaster.java (hunk 1 above)
Comment (matthiasbalke) [inline quartermaster/src/main/java/com/github/reflectoring/Quartermaster.java]: done.

File: quartermaster/src/main/java/com/github/reflectoring/infiniboard/quartermaster/dashboard/WidgetData.java (hunk 2 above)
Comment (matthiasbalke) [inline quartermaster/src/main/java/com/github/reflectoring/infiniboard/quartermaster/dashboard/WidgetData.java]: removed unused class

**Pull request merged**